src/lib/apiSchemaUtils.ts

```diff
@@ -127,11 +127,18 @@ export function getReferencedSchema(ref: string) {
     found = apiSpec.components?.schemas?.[name];
   }
 
+  if (found && "content" in found) {
+    const jsonResponse = found.content?.['application/json']
+    if (jsonResponse && jsonResponse.schema) {
+      found = jsonResponse.schema
+    }
+  }
+
   if (found && "$ref" in found) {
     throw new Error("Referenced schema can not be a reference schema itself");
   }
 
-  return found;
+  return found as OpenAPIV3.SchemaObject;
```

Comment:
I might find some time to check it out myself, but it would be great if we avoided the type assertion/cast.

I know the openapi types are very.... 'everything is a union of everything', but i would prefer this explicitly threw an exception if something isn't as expected, rather than propagate unexpectedly down.